### PR TITLE
Istio control plane overrides

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Allow the Istio Control-Plane definition as an override for the Kyma-Installer.
There is a new override for Istio chart: `kyma_istio_control_plane`.
The value of this override should be a proper YAML definition of `install.istio.io/v1alpha2:IstioControlPlane` CR.
The settings defined in the override are merged with default settings from Istio chart.

Changes proposed in this pull request:

- Introduce `kyma_istio_control_plane` override for Istio installation

**Related issue(s)**
See also #7054

### CODE REVIEW

File: resources/istio/templates/configmap-cluster.yaml [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
 metadata:
   namespace: {{ .Release.Namespace }}
-  name: kyma-cluster-config
+  name: kyma-istio-control-plane-config
 data:
   config.yaml: |-
 {{.Files.Get "files/kyma_cluster_istio_control_plane.yaml" | printf "%s" | indent 4}}
+{{ if .Values.kyma_istio_control_plane }}
+  #This additional user-provided definition is merged with the default one
+  overrides.yaml: |-
```
Comment (Tomasz-Smelcerz-SAP) [inline resources/istio/templates/configmap-cluster.yaml]: Note for reviewer: Actual size of the configmap is currently about 40KiB. This is way below the etcd object limit of 1MiB.

File: installation/resources/installer-config-local.yaml.tpl [hunk 2]
```diff
@@ -146,4 +146,366 @@ metadata:
     component: core
     kyma-project.io/installation: ""
 data:
-  kubeless.tests.enabled: "false"
\ No newline at end of file
+  kubeless.tests.enabled: "false"
+---
+apiVersion: v1
```
Comment (Tomasz-Smelcerz-SAP) [inline installation/resources/installer-config-local.yaml.tpl]: Note for reviewer: This is for test purposes only, will be removed before merge

File: resources/istio/files/job-script.sh [hunk 3]
```diff
@@ -45,8 +45,14 @@ if [ ! -z "$overrides" ]; then
...
-printf "istioctl manifest apply -f /etc/istio/config.yaml ${overrides_transformed}\n"
-istioctl manifest apply -f /etc/istio/config.yaml ${overrides_transformed}
+istio_control_plane_overrides=""
+
+if [ -f "/etc/istio/overrides.yaml" ]; then
+  istio_control_plane_overrides="-f /etc/istio/overrides.yaml"
+fi
+
+printf "istioctl manifest apply -f /etc/istio/config.yaml ${istio_control_plane_overrides} ${overrides_transformed}\n"
+istioctl manifest apply -f /etc/istio/config.yaml ${istio_control_plane_overrides} ${overrides_transformed}
```
Comment (piotrmsc) [inline resources/istio/files/job-script.sh]: To consider : if istio control plane is provided perhaps it  'old' overrides aka `overrides_transformed` should be ignored. So if user provides old helm options they are taken into account but if CP is provided then it takes precedence.  Why? there's a chance that merge of values will not properly be applied.

File: resources/istio/templates/configmap-cluster.yaml [hunk 4]
```diff
@@ -1,8 +1,15 @@
+{{ if not .Values.global.isLocalEnv }}
```
Comment (piotrmsc) [inline resources/istio/templates/configmap-cluster.yaml]: nit: not needed TBH

File: installation/resources/installer-config-local.yaml.tpl [hunk 5]
```diff
@@ -146,4 +146,5 @@ metadata:
     component: core
     kyma-project.io/installation: ""
 data:
-  kubeless.tests.enabled: "false"
\ No newline at end of file
+  kubeless.tests.enabled: "false"
+---
```
Comment (Demonsthere) [inline installation/resources/installer-config-local.yaml.tpl]: Looks like a leftover change :)

**Pull request merged**